api/handlers/app_test.go

```diff
@@ -1772,6 +1774,113 @@ var _ = Describe("App", func() {
 			})
 		})
 	})
+
+	Describe("DELETE /v3/apps/:guid/processes/:process/instances/:instance", func() {
+		When("the app has a single instance in the default (web) process", func() {
+			BeforeEach(func() {
+				processRepo.ListProcessesReturns([]repositories.ProcessRecord{
+					{
+						GUID:             "process-1-guid",
+						SpaceGUID:        spaceGUID,
+						AppGUID:          appGUID,
+						Type:             "web",
+						DesiredInstances: 1,
+					},
+				}, nil)
+				req = createHttpRequest("DELETE", "/v3/apps/"+appGUID+"/processes/web/instances/0", nil)
+			})
+
+			It("restarts the instance", func() {
+				Expect(rr).To(HaveHTTPStatus(http.StatusNoContent))
+				_, actualAuthInfo, _ := appRepo.GetAppArgsForCall(0)
+				Expect(actualAuthInfo).To(Equal(authInfo))
+			})
+		})
+		When("the app has a single instance in a non-default process", func() {
+			BeforeEach(func() {
+				req = createHttpRequest("DELETE", "/v3/apps/"+appGUID+"/processes/worker/instances/0", nil)
+				processRepo.ListProcessesReturns([]repositories.ProcessRecord{
+					{
+						GUID:             "process-1-guid",
+						SpaceGUID:        spaceGUID,
+						AppGUID:          appGUID,
+						Type:             "worker",
+						DesiredInstances: 1,
+					},
+				}, nil)
+			})
+
+			It("restarts the instance", func() {
+				Expect(rr).To(HaveHTTPStatus(http.StatusNoContent))
+				_, actualAuthInfo, _ := appRepo.GetAppArgsForCall(0)
+				Expect(actualAuthInfo).To(Equal(authInfo))
+			})
+		})
+		When("the app does not exist", func() {
+			BeforeEach(func() {
+				req = createHttpRequest("DELETE", "/v3/apps/boom/processes/web/instances/0", nil)
+				appRepo.GetAppReturns(repositories.AppRecord{}, apierrors.NewForbiddenError(nil, repositories.AppResourceType))
+			})
+
+			It("returns an error", func() {
+				expectNotFoundError("App")
+			})
+		})
+		When("the app exists but the process does not", func() {
```

Comment:
The app exists by default so it would be stylistically better to omit this from the `When`. We already have a `When` for the case where the app does not exists. So this `When` can be rephrazed as `When("the process does not exists")`